src/lib/one-graph/graphql-helpers.js

```diff
@@ -0,0 +1,533 @@
+const dotProp = require('dot-prop')
+const {
+  TypeInfo,
+  getNamedType,
+  isEnumType,
+  isInputObjectType,
+  isInterfaceType,
+  isListType,
+  isNonNullType,
+  isObjectType,
+  isUnionType,
+  isWrappingType,
+  parseType,
+  print,
+  typeFromAST,
+  visit,
+  visitWithTypeInfo,
+} = require('graphql')
+
+const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1)
+
+const gatherAllReferencedTypes = (schema, query) => {
+  const types = new Set([])
+  const typeInfo = new TypeInfo(schema)
+  visit(
+    query,
+    visitWithTypeInfo(typeInfo, {
+      enter: () => {
+        const fullType = typeInfo.getType()
+        if (fullType) {
+          const typ = getNamedType(fullType)
+          if (typ) types.add(typ.name.toLocaleLowerCase().replace('oneme', ''))
```

Comment:
A comment would be nice to explains what's going on here

Reply:
I'm planning on extracting all of this to its own package shortly after this PR, but added a comment in the meantime